### DISCUSSION TITLE
Wrong formatting of `kitty.conf` with `util.dot`

### DISCUSSION THREAD

**Discussion by ChrisGVE** [LazyVim/LazyVim#4968] [Q&A]
According to the LazyVim [page](https://www.lazyvim.org/extras/util/dot), the `util.dot` assigns the `kitty.conf` file as a `bash` file, but this creates problems with linting:
<img width="1058" alt="image" src="https://github.com/user-attachments/assets/2960a87d-1a6b-46a2-b243-551026d828a0">

and when autoformatting is triggered, the same block becomes
<img width="1047" alt="image" src="https://github.com/user-attachments/assets/ac6b5f49-2202-4275-a728-ee9ec5801fb0">

This is wrong. When editing the file, I always have to disable the formatting, but is there a solution to this?

I tried adding this `lazy.lua` to my root project, but it does not help.

```lua
-- Disable autoformat for conf files
vim.api.nvim_create_autocmd({ "FileType" }, {
  pattern = "conf",
    callback = function(opts)
      vim.g.autoformat = false
    end,
})

vim.filetype.add({
  pattern = {
    [".*/kitty/.+%.conf"] = "text",
  },
})

return {}
```

Does anyone know how to rectify this, or should I open a bug report?

**Comment by dpetka2001** [comment]
This are diagnostics not formatting. Your `.lazy.lua` should look like
```lua
vim.diagnostic.enable(false)

return {}
```

**Comment by ChrisGVE** [comment]
The problem is with both: I have diagnostics, but if you look at the difference between the two screenshots, you will see that there is also reformatting errors.

Kitty allows you to map a sequence of keys to an action with this syntax: `key1>key2 [action]`. However, the `>` is interpreted as a redirection since the file is considered a bash file, which is the cause of the problem.

On the other hand, putting `vim.diagnostic.enable(false)`, as you suggest, will act for all my .dot files and not just for `kitty.conf`, which is not desirable.

**Reply by dpetka2001** [reply]
`vim.diagnostic.enable()` can also take a `bufnr` option, so if you created an autocmd with `pattern` to match the `kitty.conf`, you should be able to isolate the behavior on only the desired buffer. 

Other than that, you would have to ping the maintainer if you'd like more insight as to the choice of `kitty.conf` being considered a `bash` filetype. You should also look into `bashls` docs to see if it provides you an option to disable diagnostics per project.

**Reply by dpetka2001** [reply]
Also I tried creating a map like you do, but I don't get any diagnostics as in your first picture.
![2024-12-07_19-17](https://github.com/user-attachments/assets/c8026c46-94c3-43af-8eb5-f50020157c1b)
And `bashls` is also attached in my case
![2024-12-07_19-19](https://github.com/user-attachments/assets/ff3db30a-6078-4975-8a39-60a22eabfc84)

**Reply by dpetka2001** [reply]
You can also use in your `.lazy.lua`
```lua
vim.api.nvim_create_autocmd("BufReadPost", {
	pattern = "*",
	callback = function()
		vim.filetype.add({
			pattern = {
				[".*/kitty/.+%.conf"] = "conf",
			},
		})
	end,
})

return {}
```
so that `kitty.conf` is not recognized as a `bash` filetype.

**Reply by ChrisGVE** [reply]
Wow, this is really odd. Thanks for your time investigating the problem. My `LspInfo` gives the same result as yours, but with the outcome you can see above. But your snippet does work better than what I tried earlier. Thanks again!

**Reply by folke** [reply]
See my other comment. For some reason I also didnt have that issue with diagnostics and formatting.
Either way, should be fixed for you as well.

**Comment by folke** [comment]
I just pushed an update to LazyVim.
Filetype will now be `kitty`, but it will still use the bash treesitter lang for highlights. (which are far better than `conf`)

Use LazyVim `main`

**Reply by ChrisGVE** [reply]
Perfect! Thanks a lot, Folke. Indeed, the highlights are much better. There is no more problem with re-formatting now.
